gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -440,6 +442,22 @@ public synchronized boolean requestTargetNumberOfContainers(int numContainers, S
     return true;
   }
 
+  public synchronized void requestNewContainers(StaffingDeltas deltas) {
+    deltas.getPerProfileDeltas().forEach(profileDelta -> {
+      if (profileDelta.getDelta() > 0) {
+        LOGGER.info("Requesting {} new containers for profile {}", profileDelta.getDelta(), profileDelta.getProfile());
+        requestContainersForWorkerProfile(profileDelta.getProfile(), profileDelta.getDelta());
+      }
+      // TODO: Decide how to handle negative deltas
+    });
+  }
+
+  private synchronized void requestContainersForWorkerProfile(WorkerProfile workerProfile, int numContainers) {
+    int containerMemoryMbs = workerProfile.getConfig().getInt(GobblinYarnConfigurationKeys.CONTAINER_MEMORY_MBS_KEY);
+    int containerCores = workerProfile.getConfig().getInt(GobblinYarnConfigurationKeys.CONTAINER_CORES_KEY);
+    requestContainers(numContainers, Resource.newInstance(containerMemoryMbs, containerCores));
```

Comment:
this is good for requesting the container, *AND* you'll elsewhere additionally need to supply the JVM props for the `GobblinTemporalYarnTaskRunner::main` to launch with.  the props to use are those of `workerProfile.getConfig()`